impeller/renderer/backend/vulkan/allocator_vk.cc

```diff
@@ -132,12 +132,27 @@ static constexpr vk::ImageUsageFlags ToVKImageUsageFlags(PixelFormat format,
 
   if (usage & static_cast<TextureUsageMask>(TextureUsage::kShaderRead)) {
     vk_usage |= vk::ImageUsageFlagBits::eSampled;
+    // unmark the usage as transient if it is used as a sampled image
+    // See: https://github.com/flutter/flutter/issues/121633
+    if (mode == StorageMode::kDeviceTransient) {
+      vk_usage &= ~vk::ImageUsageFlagBits::eTransientAttachment;
+    }
   }
 
   if (usage & static_cast<TextureUsageMask>(TextureUsage::kShaderWrite)) {
     vk_usage |= vk::ImageUsageFlagBits::eStorage;
+    // unmark the usage as transient if it is used as a storage image
+    // See: https://github.com/flutter/flutter/issues/121633
+    if (mode == StorageMode::kDeviceTransient) {
+      vk_usage &= ~vk::ImageUsageFlagBits::eTransientAttachment;
+    }
   }
 
+  // TODO (https://github.com/flutter/flutter/issues/121634):
+  // Add transfer usage flags to support blit passes
+  vk_usage |= vk::ImageUsageFlagBits::eTransferSrc |
```

Comment:
This [disables framebuffer compression for all images](https://arm-software.github.io/vulkan_best_practice_for_mobile_developers/samples/performance/afbc/afbc_tutorial.html). We should file a bug to fix it if we leave this in.

Reply:
I will reword https://github.com/flutter/flutter/issues/121634 to remove this.